packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -368,6 +368,15 @@ export default util.createRule({
      * Gets a normalized representation of the given MemberExpression
      */
     function getMemberExpressionText(node: TSESTree.MemberExpression): string {
+      if (
+        [
+          AST_NODE_TYPES.TSAsExpression,
+          AST_NODE_TYPES.AwaitExpression,
+          AST_NODE_TYPES.NewExpression,
+        ].includes(node.object.type)
+      ) {
+        return '';
+      }
```

Comment:
This feels a bit like a hack since typescript doesn't seem to think they are assignable to `MemberExpression`
`if (node.object.type === AST_NODE_TYPES.TSAsExpression) {}` shows a typescript error, but with Array#includes it does not

Reply:
It is a bit of a hack 😄 but the root cause is a bug in our typings, I think. A node's member _can_ be an `await` or `new`. [[typescript-eslint playground](https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&showAST=es&code=MYewdgzgLgBApgDwIYFsAOAbOMC8MDeAvjEhCWAJ4DcAUDQBRIDuSAlrIqpnAJQB0aAE4g0tRi3YwwcJvGTos-ISNo1EaEINj4YhKkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkeAhpgObp4VTolHoAebANOBqQSeTMmh4AcgFdUqAMIALdNADWZSmJnohAXxB6gA)]


```ts
const example = {} as any;

(await example).prop;
(await new example).prop;
```

For now, you can use a `// @ts-expect-error` and link to #6239